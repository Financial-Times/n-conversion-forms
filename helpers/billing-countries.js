import { billingCountries } from 'n-common-static-data';

export { billingCountries };
