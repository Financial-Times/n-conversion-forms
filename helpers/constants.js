const printRegions = {
	cemeaV1: 'CEMEA_V1',
	cemeaV2: 'CEMEA_V2',
	apac: 'APAC',
};

const addressLine1AutoComplete = 'address-line1';
const addressLine2AutoComplete = 'address-line2';
const addressLine3AutoComplete = 'address-line3';

module.exports = {
	printRegions,
	addressLine1AutoComplete,
	addressLine2AutoComplete,
	addressLine3AutoComplete,
};
