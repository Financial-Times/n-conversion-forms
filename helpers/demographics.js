const { demographics } = require('n-common-static-data');

export { demographics };
