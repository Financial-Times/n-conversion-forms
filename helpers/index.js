module.exports = {
	'ncf-countries': require('./ncf-countries'),
	'ncf-common-data': require('./ncf-common-data'),
	cemeaV1ISO: require('./supportedCountries').cemeaV1ISO,
	cemeaV2ISO: require('./supportedCountries').cemeaV2ISO,
	apacISO: require('./supportedCountries').apacISO,
	identifyFTShippingZone: require('./supportedCountries')
		.identifyFTShippingZone,
	countriesSupported: require('./supportedCountries').countriesSupported,
	countriesSupportedISO: require('./supportedCountries').countriesSupportedISO,
	supportedPostcodeValidators: require('./supportedPostcodeValidators')
		.supportedPostcodeValidators,
	allSupportedPostcodeExamples: require('./supportedPostcodeExamples')
		.allSupportedPostcodeExamples,
	demographics: require('./demographics').demographics,
	billingCountries: require('./billing-countries').billingCountries,
};
