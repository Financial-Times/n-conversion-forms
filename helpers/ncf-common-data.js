const _get = require('lodash.get');
const COMMON_DATA = require('n-common-static-data');

module.exports = function ({ hash, fn }) {
	const context = Object.assign({}, { [hash.export]: _get(COMMON_DATA, hash.import) }, this);
	return fn(context);
};
