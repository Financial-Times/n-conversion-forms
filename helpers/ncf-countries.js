const { countries } = require('n-common-static-data').billingCountries;

const MINIMUM_TO_GROUP = 50;
const MINIMUM_TO_SHOW_GROUPS = 2;

module.exports = function ({ hash = {}, fn }) {
	const filter = Array.isArray(hash.filterList) ? hash.filterList : false;
	const value = hash.value;

	let data = countries;

	// Only show countries in the filter
	if (filter) {
		data = data.filter(item => filter.includes(item.code));
	}

	// Select the country
	if (value) {
		data = data.map(item => {
			item.selected = item.code === value;
			return item;
		});
	}

	// Group countries
	if (data.length >= MINIMUM_TO_GROUP) {
		data = groupCountries(data);
	}

	const context = Object.assign({ countries: data }, this);
	return fn(context);
};

/**
 * Produce a frequently used option group
 * @param {Array} countries
 * @return {Array}
 */
function groupCountries (countries) {
	const frequentlyUsed = ['GBR', 'USA', 'JPN', 'FRA', 'CAN'];

	const frequentlyUsedCountries = countries
		.filter(item => frequentlyUsed.includes(item.code))
		.map(item => Object.assign({}, item))
		.sort((a, b) => frequentlyUsed.indexOf(a.code) - frequentlyUsed.indexOf(b.code));

	// Not worth showing return standard countries list
	if (frequentlyUsedCountries.length < MINIMUM_TO_SHOW_GROUPS) {
		return countries;
	}

	// If selected country is in frequently used only leave that one selected
	// otherwise the user is shown the lowest selected option
	const selected = frequentlyUsedCountries.find(item => item.selected);
	if (selected) {
		countries.forEach(item => item.selected = false);
	}

	return [
		{ label: 'Frequently Used', countries: frequentlyUsedCountries },
		{ label: 'Alphabetical', countries }
	];
}
