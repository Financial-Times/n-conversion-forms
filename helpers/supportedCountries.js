const supportedCountriesMasterList = {
	cemeaV1: {
		AUT: 'Austria',
		BEL: 'Belgium',
		BGR: 'Bulgaria',
		HRV: 'Croatia',
		CZE: 'Czechia',
		DNK: 'Denmark',
		FIN: 'Finland',
		DEU: 'Germany',
		GRC: 'Greece',
		HUN: 'Hungary',
		ISL: 'Iceland',
		ITA: 'Italy',
		LIE: 'Liechtenstein',
		LTU: 'Lithuania',
		NLD: 'Netherlands',
		NOR: 'Norway',
		POL: 'Poland',
		PRT: 'Portugal',
		QAT: 'Qatar',
		ROU: 'Romania',
		RUS: 'Russian Federation',
		SVK: 'Slovakia',
		SVN: 'Slovenia',
		ZAF: 'South Africa',
		ESP: 'Spain',
		SWE: 'Sweden',
		CHE: 'Switzerland'
	},
	cemeaV2: { FRA: 'France', LUX: 'Luxembourg', MCO: 'Monaco' },
	apac: {
		CHN: 'China',
		HKG: 'Hong Kong',
		IND: 'India',
		IDN: 'Indonesia',
		JPN: 'Japan',
		KOR: 'Korea, Republic of',
		MYS: 'Malaysia',
		PHL: 'Philippines',
		SGP: 'Singapore',
		TWN: 'Taiwan, Province of China',
		THA: 'Thailand'
	},
	GBR: 'United Kingdom',
	USA: 'United States',
	CAN: 'Canada'
};


const cemeaV1ISO = Object.keys(supportedCountriesMasterList.cemeaV1);
const cemeaV2ISO = Object.keys(supportedCountriesMasterList.cemeaV2);
const apacISO = Object.keys(supportedCountriesMasterList.apac);
const countriesSupported = Object.assign({}, ...(function _flatten (o) { return [].concat(...Object.keys(o).map(k => typeof o[k] === 'object' ? _flatten(o[k]) : ({[k]: o[k]})));}(supportedCountriesMasterList)));


module.exports = { cemeaV1ISO, cemeaV2ISO, apacISO, countriesSupported };
