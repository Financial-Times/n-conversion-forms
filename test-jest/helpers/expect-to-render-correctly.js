import ReactDOMServer from 'react-dom/server';

export const expectToRenderCorrectly = {
	toRenderCorrectly(reactComponent, props) {
		// Render the React component with props.
		const ReactComponentRendered = ReactDOMServer.renderToStaticMarkup(
			reactComponent(props)
		);

		// eslint-disable-next-line jest/no-standalone-expect
		expect(ReactComponentRendered).toMatchSnapshot();

		// This point is reached when the above assertions were successful.
		// The test should therefore always pass, that means it needs to be
		// `true` when used normally, and `false` when `.not` was used.
		return {
			pass: !this.isNot,
		};
	},
};
