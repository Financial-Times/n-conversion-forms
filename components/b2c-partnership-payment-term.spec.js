import { B2cPartnershipPaymentTerm } from './index';
import { expectToRenderCorrectly } from '../test-jest/helpers/expect-to-render-correctly';

expect.extend(expectToRenderCorrectly);

const defaultProps = {
	displayName: 'Digital FT + The Washington Post',
	partnerOffer: {
		duration: '90-day',
		name: 'All-Access Digital',
		vendor: 'The Washington Post'
	},
	offerType: 'DIGITAL',
	options: [
		{
			name: 'annual',
			price: '£299.99',
			value: 299.99,
			isTrial: false,
			discount: null,
			selected: true
		}
	]
};

describe('B2cPartnershipPaymentTerm', () => {
	it('render with minimum props required', () => {
		const props = {
			...defaultProps,
		};

		expect(B2cPartnershipPaymentTerm).toRenderCorrectly(props);
	});

	['annual', 'quarterly', 'monthly'].forEach(type => {
		it(`renders with ${type} option`, () => {
			const props = {
				...defaultProps,
				options: [{
					name: type,
					value: type,
					price: '£20.00',
					selected: true
				}]
			};

			expect(B2cPartnershipPaymentTerm).toRenderCorrectly(props);
		});
	});
});
