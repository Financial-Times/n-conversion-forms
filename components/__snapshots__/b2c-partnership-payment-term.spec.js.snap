// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`B2cPartnershipPaymentTerm render with minimum props required 1`] = `
<div id="paymentTermField"
     class="o-forms__group ncf__payment-term ncf__b2c-partnership-payment-term"
>
  <div class="ncf__payment-term__item o-forms-input--radio-round">
    <input type="radio"
           id="299.99"
           name="paymentTerm"
           value="299.99"
           checked
    >
    <label for="299.99"
           class="o-forms-input__label ncf__payment-term__label"
    >
      <span class="ncf__payment-term__title">
        £299.99 - Digital FT + The Washington Post
      </span>
      <div class="ncf__b2c-partnership-payment-term__description">
        <p>
          Includes:
          <br>
          1 year Digital subscription to the Financial Times
          <br>
          90-day All-Access Digital subscription to The Washington Post
        </p>
      </div>
    </label>
  </div>
  <div class="ncf__payment-term__legal">
    <p>
      We will automatically renew your subscription using the payment method provided unless you cancel before your renewal date.
    </p>
    <p>
      We will notify you at least 14 days in advance of any changes to the price in your subscription that would apply upon next renewal. Find out more about our cancellation policy in our
      <a class="ncf__link--external"
         href="https://help.ft.com/help/legal-privacy/terms-conditions/"
         title="FT Legal Terms and Conditions help page"
         target="_blank"
         rel="noopener noreferrer"
      >
        Terms &amp; Conditions
      </a>
      .
    </p>
  </div>
</div>
`;

exports[`B2cPartnershipPaymentTerm renders with annual option 1`] = `
<div id="paymentTermField"
     class="o-forms__group ncf__payment-term ncf__b2c-partnership-payment-term"
>
  <div class="ncf__payment-term__item o-forms-input--radio-round">
    <input type="radio"
           id="annual"
           name="paymentTerm"
           value="annual"
           checked
    >
    <label for="annual"
           class="o-forms-input__label ncf__payment-term__label"
    >
      <span class="ncf__payment-term__title">
        £20.00 - Digital FT + The Washington Post
      </span>
      <div class="ncf__b2c-partnership-payment-term__description">
        <p>
          Includes:
          <br>
          1 year Digital subscription to the Financial Times
          <br>
          90-day All-Access Digital subscription to The Washington Post
        </p>
      </div>
    </label>
  </div>
  <div class="ncf__payment-term__legal">
    <p>
      We will automatically renew your subscription using the payment method provided unless you cancel before your renewal date.
    </p>
    <p>
      We will notify you at least 14 days in advance of any changes to the price in your subscription that would apply upon next renewal. Find out more about our cancellation policy in our
      <a class="ncf__link--external"
         href="https://help.ft.com/help/legal-privacy/terms-conditions/"
         title="FT Legal Terms and Conditions help page"
         target="_blank"
         rel="noopener noreferrer"
      >
        Terms &amp; Conditions
      </a>
      .
    </p>
  </div>
</div>
`;

exports[`B2cPartnershipPaymentTerm renders with monthly option 1`] = `
<div id="paymentTermField"
     class="o-forms__group ncf__payment-term ncf__b2c-partnership-payment-term"
>
  <div class="ncf__payment-term__item o-forms-input--radio-round">
    <input type="radio"
           id="monthly"
           name="paymentTerm"
           value="monthly"
           checked
    >
    <label for="monthly"
           class="o-forms-input__label ncf__payment-term__label"
    >
      <span class="ncf__payment-term__title">
        £20.00 - Digital FT + The Washington Post
      </span>
      <div class="ncf__b2c-partnership-payment-term__description">
        <p>
          Includes:
          <br>
          1 month Digital subscription to the Financial Times
          <br>
          90-day All-Access Digital subscription to The Washington Post
        </p>
      </div>
    </label>
  </div>
  <div class="ncf__payment-term__legal">
    <p>
      We will automatically renew your subscription using the payment method provided unless you cancel before your renewal date.
    </p>
    <p>
      We will notify you at least 14 days in advance of any changes to the price in your subscription that would apply upon next renewal. Find out more about our cancellation policy in our
      <a class="ncf__link--external"
         href="https://help.ft.com/help/legal-privacy/terms-conditions/"
         title="FT Legal Terms and Conditions help page"
         target="_blank"
         rel="noopener noreferrer"
      >
        Terms &amp; Conditions
      </a>
      .
    </p>
  </div>
</div>
`;

exports[`B2cPartnershipPaymentTerm renders with quarterly option 1`] = `
<div id="paymentTermField"
     class="o-forms__group ncf__payment-term ncf__b2c-partnership-payment-term"
>
  <div class="ncf__payment-term__item o-forms-input--radio-round">
    <input type="radio"
           id="quarterly"
           name="paymentTerm"
           value="quarterly"
           checked
    >
    <label for="quarterly"
           class="o-forms-input__label ncf__payment-term__label"
    >
      <span class="ncf__payment-term__title">
        £20.00 - Digital FT + The Washington Post
      </span>
      <div class="ncf__b2c-partnership-payment-term__description">
        <p>
          Includes:
          <br>
          3 month Digital subscription to the Financial Times
          <br>
          90-day All-Access Digital subscription to The Washington Post
        </p>
      </div>
    </label>
  </div>
  <div class="ncf__payment-term__legal">
    <p>
      We will automatically renew your subscription using the payment method provided unless you cancel before your renewal date.
    </p>
    <p>
      We will notify you at least 14 days in advance of any changes to the price in your subscription that would apply upon next renewal. Find out more about our cancellation policy in our
      <a class="ncf__link--external"
         href="https://help.ft.com/help/legal-privacy/terms-conditions/"
         title="FT Legal Terms and Conditions help page"
         target="_blank"
         rel="noopener noreferrer"
      >
        Terms &amp; Conditions
      </a>
      .
    </p>
  </div>
</div>
`;
